test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -7186,6 +7186,27 @@ BOOST_AUTO_TEST_CASE(no_nonpayable_circumvention_by_modifier)
 	BOOST_CHECK_EQUAL(balanceAt(m_contractAddress), 0);
 }
 
+BOOST_AUTO_TEST_CASE(shift_constant_left)
+{
+	char const* sourceCode = R"(
+		contract C {
+			uint public a = 0x42 << 8;
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+	BOOST_CHECK(callContractFunction("a()") == encodeArgs(u256(0x4200)));
+}
+
+BOOST_AUTO_TEST_CASE(shift_constant_right)
+{
+	char const* sourceCode = R"(
+		contract C {
+			uint public a = 0x4200 >> 8;
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+	BOOST_CHECK(callContractFunction("a()") == encodeArgs(u256(0x42)));
+}
```

Comment:
`SAR` is not supported: in the implementation above negative left value is a type error. Also see the original PR #527.